naucse/templates.py

```diff
@@ -187,5 +187,10 @@ def format_date_range(start_and_end):
         else:
             parts += ['{start.day}.']
         parts += [' – ']
-        parts += [format_date(end)]
+    parts += [format_date(end)]
     return ''.join(parts).format(start=start, end=end)
+
+@template_filter()
+def monthname(number):
```

Comment:
That's a reasonable thing to say. But I don't trust locales.

In practice I found locales frequently solve either too much or too litle. See for example naucse's function for formatting date ranges – building that on top of a locale "format a date for me" function would probably turn out quite fragile, so anyway you end up with some things based on locale and some hard-coded, which means some explicit and some implicit, and you anyway need to take each language into account separately, and... it's just a mess.

Also, it's one more dependency: cs_CZ needs to be available on the target system. How do I ensure that? How do I test it? And, `setlocale` is not thread-safe. Using it would rule out the possibility of running the Czech and Slovak version on one (threaded) server.

The flip side is that the site can only be translated by programmers (or a team that has a programmer). And I am fine with that, at least for now.

Reply:
Makes sense. As I said, I was just curious.